plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStore.php

```diff
@@ -2918,16 +2919,31 @@ public function update_meta( &$object, $meta ) {
 	 *
 	 * @param WC_Abstract_Order $order Order object.
 	 * @param \WC_Meta_Data     $meta  Metadata object.
+	 *
+	 * @return bool True if changes were applied, false otherwise.
 	 */
 	protected function after_meta_change( &$order, $meta ) {
-		$current_date_time = new \WC_DateTime( current_time( 'mysql', 1 ), new \DateTimeZone( 'GMT' ) );
 		method_exists( $meta, 'apply_changes' ) && $meta->apply_changes();
 		$this->clear_caches( $order );
 
 		// Prevent this happening multiple time in same request.
-		if ( $order->get_date_modified() < $current_date_time && empty( $order->get_changes() ) ) {
+		if ( $this->should_save_after_meta_change( $order ) ) {
 			$order->set_date_modified( current_time( 'mysql' ) );
 			$order->save();
+			return true;
 		}
+		return false;
+	}
+
+	/**
+	 * Helper function to check whether the modified date needs to be updated after a meta save.
+	 *
+	 * @param WC_Order $order Order object.
+	 *
+	 * @return bool Whether the modified date needs to be updated.
+	 */
+	private function should_save_after_meta_change( $order ) {
+		$current_date_time = new \WC_DateTime( current_time( 'mysql', 1 ), new \DateTimeZone( 'GMT' ) );
+		return $order->get_date_modified() < $current_date_time && empty( $order->get_changes() );
```

Comment:
Nice that you extracted the test to a separate function, makes the code clearer! 👍 

This test feels a bit unclear to me, so let me try to explain how I understand it: 
- in case there are any changes for `$order`, we assume the order will be saved at some later point
- we want to make sure the `$order` has correct date_modifed set and don't want to updated it over and over when saving multiple meta data, so we compare the current time to the order modified time. If the order was already saved during this second, we won't do it again

Is that correct?

So e.g. when calling multiple times during a request that takes a couple of seconds, order's date modified will only be saved once per second, so that the latest timestamp will be stored as order's modified date, e.g.

s_1, set_date_modified
s_1, skip save
s_1, skip save
s_2, set_date_modified
s_2, skip save

Okay, I guess that should work correctly. Still, would be nice to add a bit of explanation to make it easier to decipher what's happening here and why.

Reply:
Yeah this is correct, I have updated the documentation here: https://github.com/woocommerce/woocommerce/pull/40129/commits/252c349fd25309ad657aeba3e326d863132a8831